File: bin/bold_my_name.py
#!/usr/bin/env python3
import sys
import re
import os

# 定义要加粗的名字列表
MY_NAMES = [
    "Fu, X.", "Xiaokang Fu", "Fu, Xiaokang", "付小康",
    "Fu†, X.", "Fu†, Xiaokang"
]

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        for name in MY_NAMES:
            # 使用 re.escape 转义特殊字符，并构建正则
            # 使用 g 标志是 JS 的，Python 中直接用 sub 替换所有匹配
            regex = re.compile(re.escape(name))
            content = regex.sub(f"<strong>{name}</strong>", content)
            
        if content != original_content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Bolded names in: {filepath}")
            
    except Exception as e:
        print(f"Error processing {filepath}: {e}")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python3 bold_my_name.py <file1> <file2> ...")
        sys.exit(1)
        
    for file_path in sys.argv[1:]:
        process_file(file_path)

File: bin/bold_my_name.pl
#!/usr/bin/perl

use strict;
use warnings;

my @my_names = (
    "Fu, X.", "Xiaokang Fu", "Fu, Xiaokang", "付小康",
    # 如果有带dagger的显示，也在这里添加
    "Fu†, X.", "Fu†, Xiaokang", "Fu†, 付小康"
);

# 构建替换的正则表达式
my $name_regex = join("|", map { quotemeta($_) } @my_names);
$name_regex = qr/($name_regex)/;

# 从命令行参数获取文件列表
my @files = @ARGV;

foreach my $file (@files) {
    next unless -f $file; # 确保是文件
    
    my $content;
    open(my $fh, '<:encoding(UTF-8)', $file) or die "Cannot open $file: $!";
    {
        local $/; # 切换到“段落模式”，一次读完整个文件
        $content = <$fh>;
    }
    close $fh;

    # 进行替换，使用 <strong> 标签加粗
    $content =~ s/$name_regex/<strong>$1<\/strong>/g;

    open(my $fh_out, '>:encoding(UTF-8)', $file) or die "Cannot write to $file: $!";
    print $fh_out $content;
    close $fh_out;
    
    print "Processed $file\n";
}
